Tests/CacheAdvanceTests/SQLitePerformanceComparisonTests.swift

```diff
@@ -0,0 +1,307 @@
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SQLite3
+import XCTest
+
+@testable import CacheAdvance
+
+final class SQLitePerformanceComparisonTests: XCTestCase {
+
+    // MARK: XCTestCase
+
+    override func setUp() {
+        super.setUp()
+
+        // Delete the existing cache.
+        try? FileManager.default.removeItem(at: testFileLocation)
+    }
+
+    // MARK: Behavior Tests
+
+    func test_append_sqlite_fillableCache_canMaintainMaxCount() {
+        let cache = SQLiteCache<TestableMessage>(
+            location: testFileLocation,
+            maxMessageCount: Int32(LorumIpsum.messages.count),
+            shouldOverwriteMessages: false)
+        for message in LorumIpsum.messages + LorumIpsum.messages {
+            cache.appendMessage(message)
+        }
+
+        XCTAssertEqual(cache.messages().count, LorumIpsum.messages.count)
+    }
+
+    func test_messages_sqlite_fillableCache_canReadInsertedMessages() {
+        let cache = SQLiteCache<TestableMessage>(
+            location: testFileLocation,
+            maxMessageCount: Int32(LorumIpsum.messages.count),
+            shouldOverwriteMessages: true)
+        for message in LorumIpsum.messages {
+            cache.appendMessage(message)
+        }
+
+        XCTAssertEqual(cache.messages(), LorumIpsum.messages)
+    }
+
+    func test_append_sqlite_overwritingCache_canMaintainMaxCount() {
+        let cache = SQLiteCache<TestableMessage>(
+            location: testFileLocation,
+            maxMessageCount: Int32(LorumIpsum.messages.count),
+            shouldOverwriteMessages: true)
+        for message in LorumIpsum.messages + LorumIpsum.messages {
+            cache.appendMessage(message)
+        }
+
+        XCTAssertEqual(cache.messages().count, LorumIpsum.messages.count)
+    }
+
+    func test_messages_sqlite_overwritingCache_canReadInsertedMessages() {
+        let cache = SQLiteCache<TestableMessage>(
+            location: testFileLocation,
+            maxMessageCount: Int32(LorumIpsum.messages.count),
+            shouldOverwriteMessages: true)
+        for message in LorumIpsum.messages {
+            cache.appendMessage(message)
+        }
+
+        XCTAssertEqual(cache.messages(), LorumIpsum.messages)
+    }
+
+    // MARK: Performance Tests
+
+    func test_performance_createDatabaseAndAppendSingleMessage() {
+        measure {
+            // Delete any existing database.
+            try? FileManager.default.removeItem(at: testFileLocation)
+
+            let cache = SQLiteCache<TestableMessage>(
+                location: testFileLocation,
+                maxMessageCount: Int32(LorumIpsum.messages.count),
+                shouldOverwriteMessages: false)
+
+            cache.appendMessage("test message")
+        }
+    }
+
+    func test_performance_sqlite_append_fillableCache() {
+         // Create a cache that won't run out of room over multiple test runs.
```

Comment:
`measure` is called 10 times for wall-clock tests. For memory tests, it's called 5 times. Other tests may yet be different, but I haven't confirmed.

`measure` has called the block 10 times for as long as I've been using this function (at least 6 years). I wish this were documented rather than observed, but, it is running 10 times and then producing an average. You can see this in the CI output. Example (from a local run):
```
Test Case '-[CacheAdvanceTests.CacheAdvanceTests test_performance_messages_overwritingCache]' measured [Time, seconds] average: 0.007, relative standard deviation: 7.324%, values: [0.007642, 0.006779, 0.006662, 0.007822, 0.007862, 0.006959, 0.006448, 0.006547, 0.006742, 0.006727], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100
```